src/org/rascalmpl/values/parsetrees/TreeAdapter.java

```diff
@@ -266,6 +270,41 @@ else if (ProductionAdapter.isRegular(prod)) {
 						return null;
 				}
 			}
+			else if (ProductionAdapter.isError(prod)) {
+				int dot = ProductionAdapter.getErrorDot(prod);
+				IConstructor eprod = ProductionAdapter.getErrorProd(prod);
+				IList syms = ProductionAdapter.getSymbols(eprod);
+				int index = SymbolAdapter.indexOfLabel(syms, field);
+
+				if (index != -1) {
+					IConstructor sym = (IConstructor) syms.get(index);
+					sym = SymbolAdapter.stripLabelsAndConditions(sym);
+
+					if (dot <= index) {
+						// we have parsed the field so we can just return it.
+						// this is a likely scenario
+						return new FieldResult(sym, (ITree) tree.getArgs().get(index));
+					}
+					else {
+						// We have the right prodction and the field would be there, if we didn't recover from a parse error
+						// and are missing some of the children (including the indicated field).
+						// So we return a quasi tree that is of the right type, but it otherwise just an error tree
+						// like its parent. This tree does not have content, because we wouldn't know which of the skipped
+						// parts are meant.
+						var vf = IRascalValueFactory.getInstance();
+						// @PieterOlivier is this the right way to do this? I just need an empty tree of the right symbol.
```

Comment:
The 'skipped' prod should have a symbol as only argument. I think we should use the same symbol we use for "filled" skipped nodes. At least that is what I do when an empty skip node is created (can only happen at the end of the input).
`UPTRNodeFactory` has methods to create skipped/error nodes, using that would increase readability of this section of code.

Reply:
nice. thanks!